File: Artist/app/src/main/java/com/example/artist/ui/topalbumslisting/TopAlbumsInteractorImpl.java
package com.example.artist.ui.topalbumslisting;


import com.example.artist.models.TopAlbumsResponse;
import com.example.artist.network.TopAlbumsService;

import io.reactivex.Single;
import retrofit2.Retrofit;


public class TopAlbumsInteractorImpl implements TopAlbumsInteractor {
    Retrofit mRetrofit;

    public TopAlbumsInteractorImpl(Retrofit retrofit) {
        mRetrofit = retrofit;
    }

    @Override
    public Single<TopAlbumsResponse> getTopAlbums(String userName, int limit, String apiKey) {
        return mRetrofit.create(TopAlbumsService.class).getTopArtists(userName, limit, apiKey);
    }
}

File: Artist/app/src/main/java/com/example/artist/ui/topalbumslisting/TopAlbumsInteractorImpl.kt
package com.example.artist.ui.topalbumslisting

import com.example.artist.models.TopAlbumsResponseResult
import com.example.artist.network.TopAlbumsService
import io.reactivex.Single
import retrofit2.Retrofit

class TopAlbumsInteractorImpl(var mRetrofit: Retrofit) : TopAlbumsInteractor {
    override fun getTopAlbums(userName: String, apiKey: String): Single<TopAlbumsResponseResult> {
        return mRetrofit.create(TopAlbumsService::class.java).getTopArtists(userName, apiKey)
    }
}